modules/eval-args.nix

```diff
@@ -0,0 +1,18 @@
+# Arguments for `lib.evalModules` or `types.submoduleWith`.
+{ pkgs
+, lib
+, modules ? [ ]
+, extraSpecialArgs ? { }
+}:
+let
+  devenvModules = import ./modules.nix {
+    inherit lib pkgs;
+  };
+in
+{
+  modules = (lib.toList modules) ++ devenvModules;
+  specialArgs = {
+    modulesPath = builtins.toString ./.;
+    extraModulesPath = builtins.toString ../extra;
```

Comment:
Can you keep extraModulesDir as an alias, to make the migration easier?

Reply:
I'm certainly happy to add this alias, though I believe that this change is backward-compatible: the current module argument is `extraModulesPath`; `extraModulesDir` is only used internally.